data/mods/gen1/scripts.js

```diff
@@ -28,18 +28,18 @@ let BattleScripts = {
 	// BattlePokemon scripts.
 	pokemon: {
 		getStat(statName, unmodified) {
-			statName = toId(statName);
-			if (statName === 'hp') return this.maxhp;
-			if (unmodified) return this.stats[statName];
+			statName = /** @type {StatNameExceptHP} */(toId(statName));
+			// @ts-ignore - type checking prevents 'hp' from being passed, but we're paranoid
```

Comment:
You _can_ actually cast it as `StatName` and the throw will automatically refine it to `StatNameExceptHP`.

Flow typing is actually amazing.

Reply:
Oh wait, the argument is `StatNameExceptHP`. Nevermind, this is correct.